packages/versioning/src/validate.ts

```diff
@@ -21,20 +21,17 @@ import {
 } from "./versioning.js";
 
 export function $onValidate(program: Program) {
-  const namespaceDependencies = new Map();
+  const namespaceDependencies = new Map<Namespace | undefined, Set<Namespace>>();
 
   function addDependency(source: Namespace | undefined, target: Type | undefined) {
-    if (target === undefined || !("namespace" in target) || target.namespace === undefined) {
+    if (!target || !("namespace" in target) || !target.namespace) {
```

Comment:
Why not specifically check for undefined?

Reply:
It seemed verbose. Is there a particular reason to check for undefined?